### PR TITLE
Add SQL VM Storage Configuration Template

### PR DESCRIPTION
**Pull request by sam0227**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog
*Added new template for storage configuration for SQL VM
*
*

### CODE REVIEW

File: 101-sql-vm-new-storage/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,315 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "virtualMachineName": {
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/azuredeploy.json]: add a suitable defaultValue

File: 101-sql-vm-new-storage/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+      "defaultValue": "Standard_DS13_v2",
+      "allowedValues": [
+        "Standard_DS13_v2",
+        "Standard_DS3_v2"
+      ],
+      "metadata": {
+        "description": "The virtual machine size."
+      }
+    },
+    "exisitingVirtualNetworkName": {
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/azuredeploy.json]: need to include a resourceGroup name for the vnet as well and use it in the resourceIds

File: 101-sql-vm-new-storage/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+        "DW"
+      ],
+      "defaultValue": "General",
+      "metadata": {
+        "description": "SQL Server Workload Type"
+      }
+    },
+    "sqlDataDisksCount": {
+      "type": "int",
+      "defaultValue": 1,
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/azuredeploy.json]: add min/max values

File: 101-sql-vm-new-storage/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+    },
+    "dataPath": {
+      "type": "String",
+      "defaultValue": "F:\\SQLData",
+      "metadata": {
+        "description": "Path for SQL Data files"
+      }
+    },
+    "sqlLogDisksCount": {
+      "type": "int",
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/azuredeploy.json]: add min/max values

File: 101-sql-vm-new-storage/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+    "location": {
+      "type": "string",
+      "metadata": {
+        "description": "Location for all resources."
+      },
+      "defaultValue": "[resourceGroup().location]"
+    }
+  },
+  "variables": {
+    "networkInterfaceName": "[concat(parameters('virtualMachineName'), '-nic-', uniqueString(parameters('virtualMachineName')))]",
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/azuredeploy.json]: remove uniqueString, this need not be globally unique

File: 101-sql-vm-new-storage/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+    "networkSecurityGroupName": "[concat(parameters('virtualMachineName'), '-nsg-', uniqueString(parameters('virtualMachineName')))]",
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/azuredeploy.json]: remove uniqueString, this need not be globally unique

File: 101-sql-vm-new-storage/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+          "destinationAddressPrefix": "*",
+          "destinationPortRange": "3389"
+        }
+      }
+    ],
+    "publicIpAddressName": "[concat(parameters('virtualMachineName'), '-publicip-', uniqueString(parameters('virtualMachineName')))]",
+    "publicIpAddressType": "Dynamic",
+    "publicIpAddressSku": "Basic",
+    "diskConfigurationType": "NEW",
+    "nsgId": "[resourceId(resourceGroup().name, 'Microsoft.Network/networkSecurityGroups', variables('networkSecurityGroupName'))]",
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/azuredeploy.json]: remove resourceGroup().name

File: 101-sql-vm-new-storage/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,315 @@
...
+    "vnetId": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/',resourceGroup().name, '/providers/','Microsoft.Network/virtualNetWorks/', parameters('exisitingVirtualNetworkName'))]",
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/azuredeploy.json]: remove vnetId

File: 101-sql-vm-new-storage/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,315 @@
...
+    "subnetRef": "[concat(variables('vnetId'), '/subnets/', parameters('exisitingSubNetName'))]",
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/azuredeploy.json]: remove concat and use resourceId,

resourceId(parameters('existingVirtualNetworkResourceGroupName'), 'Microsoft.Network/virtualNetworks/subnets', parameters('exisitingVirtualNetworkName'), parameters('exisitingSubNetName')

File: 101-sql-vm-new-storage/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,315 @@
...
+    "dataDisksLuns": "[array(range(0 ,parameters('sqlDataDisksCount')))]",
+    "logDisksLuns": "[array(range(parameters('sqlDataDisksCount'), parameters('sqlLogDisksCount')))]",
+    "dataDisks": {
+      "createOption": "empty",
+      "caching": "ReadOnly",
+      "writeAcceleratorEnabled": false,
+      "id": null,
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/azuredeploy.json]: remove null properties

File: 101-sql-vm-new-storage/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,315 @@
...
+        "ipConfigurations": [
+          {
+            "name": "ipconfig1",
+            "properties": {
+              "subnet": {
+                "id": "[variables('subnetRef')]"
+              },
+              "privateIPAllocationMethod": "Dynamic",
+              "publicIpAddress": {
+                "id": "[resourceId(resourceGroup().name, 'Microsoft.Network/publicIpAddresses', variables('publicIpAddressName'))]"
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/azuredeploy.json]: remove resourceGroup().name

File: 101-sql-vm-new-storage/azuredeploy.json [hunk 12]
```diff
@@ -0,0 +1,315 @@
...
+              "name": "dataDisks",
+              "count": "[add(parameters('sqlDataDisksCount'), parameters('sqlDataDisksCount'))]",
+              "input": {
+                "lun": "[copyIndex('dataDisks')]",
+                "createOption": "[variables('dataDisks').createOption]",
+                "caching": "[if(greaterOrEquals(copyIndex('dataDisks'), parameters('sqlDataDisksCount')) ,'None', variables('dataDisks').caching )]",
+                "writeAcceleratorEnabled": "[variables('dataDisks').writeAcceleratorEnabled]",
+                "diskSizeGB": "[variables('dataDisks').diskSizeGB]",
+                "managedDisk": {
+                  "id": "[coalesce(variables('dataDisks').id, if(equals(variables('dataDisks').name, json('null')), json('null'), resourceId('Microsoft.Compute/disks', variables('dataDisks').name)))]",
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/azuredeploy.json]: remove this and use implicit disks

File: 101-sql-vm-new-storage/metadata.json [hunk 13]
```diff
@@ -0,0 +1,9 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "type": "QuickStart",
+  "itemDisplayName": "Create SQL Server Virtual Machine with performance optimized storage settings",
+  "description": "waiting for Mine's descrption",
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/metadata.json]: Get description from Mine or just copy the summary

File: 101-sql-vm-new-storage/prereqs/prereq.azuredeploy.json [hunk 14]
```diff
@@ -0,0 +1,48 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {},
+  "variables": {
+    "firstVNETName": "vnet-01",
+    "firstVNETLocation": "West US",
```
Comment (bmoore-msft) [inline 101-sql-vm-new-storage/prereqs/prereq.azuredeploy.json]: this must be resourceGroup().location for CI

File: 101-sql-vm-new-storage/azuredeploy.json (hunk 1 above)
Comment (sam0227) [reply 101-sql-vm-new-storage/azuredeploy.json]: added a defaultValue

File: 101-sql-vm-new-storage/azuredeploy.json (hunk 2 above)
Comment (sam0227) [reply 101-sql-vm-new-storage/azuredeploy.json]: added

**Pull request merged**